CRM/Utils/Mail.php

```diff
@@ -203,8 +203,13 @@ public static function send(&$params) {
       CRM_Utils_Array::value('toEmail', $params),
       FALSE
     );
-    $headers['Cc'] = CRM_Utils_Array::value('cc', $params);
-    $headers['Bcc'] = CRM_Utils_Array::value('bcc', $params);
+
+    // On some servers mail() fails when 'Cc' or 'Bcc' headers are defined but empty.
+    foreach (['Cc', 'Bcc'] as $optionalHeader) {
+      if (!empty($params[$optionalHeader])) {
```

Comment:
shouldn't this be:
> if (!empty($params[**strtolower(**$optionalHeader]))) {
since it is $params['cc'] lower case in the lines that you are replacing?

I also do not like moving away from the CRM_Utils_Array functions.

How about doing instead:
> foreach (['Cc', 'Bcc'] as $optionalHeader) {
>   $headers[$optionalHeader] = CRM_Utils_Array::value(strtolower($optionalHeader), $params);
>   if (empty($headers[$optionalHeader])) {
>     unset($headers[$optionalHeader])
>   }
> }

Reply:
@nganivet do you have any opinion about the safety of the fix?

Reply:
Right now the fix is non-functional, so needs to be corrected first. Then it would depend on the downstream library's expectations and behavior: does it require $headers['cc'] to be set, even if empty? is it creating the CC header in the outgoing email if the $headers['cc'] is not set? I have not investigated these questions.

Reply:
OK @nganivet those questions are the key ones we need to answer. This patch has changed a few times since created & I agree the current iteration needs to do a strtolower comparison - but we really need to answer the bigger question - ie. we know some mail servers fail on an empty cc / bcc but we don't know if any servers REQUIRE cc / bcc to be set EVEN IF EMPTY

Reply:
@nganivet Thanks for the feedback, I cleaned the patch up and as you spotted it no longer worked because of the lowercase issue.  I've now amended it with changes as you suggested.